src/lib/profiles/data-management/Current/SubscriptionClient.cpp

```diff
@@ -2915,6 +2915,32 @@ WEAVE_ERROR SubscriptionClient::SetUpdated(TraitUpdatableDataSink * aDataSink, P
     return err;
 }
 
+WEAVE_ERROR SubscriptionClient::ClearUpdated(TraitUpdatableDataSink * aDataSink, PropertyPathHandle aPropertyHandle)
+{
+    WEAVE_ERROR err = WEAVE_NO_ERROR;
+    TraitDataHandle dataHandle;
+    const TraitSchemaEngine * schemaEngine;
+    bool needToSetUpdateRequiredVersion = false;
+    bool isTraitInstanceInUpdate = false;
+
+    LockUpdateMutex();
+
+    schemaEngine = aDataSink->GetSchemaEngine();
+
+    err = mDataSinkCatalog->Locate(aDataSink, dataHandle);
+    SuccessOrExit(err);
+
+    isTraitInstanceInUpdate = mPendingUpdateSet.IsTraitPresent(dataHandle);
+
+    VerifyOrExit(isTraitInstanceInUpdate, WeaveLogDetail(DataManagement, "trait %d is not in update pending set, skip ClearUpdated", dataHandle));
```

Comment:
Do not put other actions in the second parameter to VeriryOrExit().  The second argument should only contain assignments to return values.

Reply:
Done, thanks